mx_benchmark.py

```diff
@@ -0,0 +1,304 @@
+#
+# ----------------------------------------------------------------------------------------------------
+#
+# Copyright (c) 2016, 2016, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+# ----------------------------------------------------------------------------------------------------
+
+import json
+import re
+import socket
+import subprocess
+import uuid
+from argparse import ArgumentParser
+
+import mx
+
+
+_benchmark_suites = {}
+
+
+class BenchmarkSuite(object):
+    def name(self):
+        """Returns the name of the suite.
+
+        Returns:
+            basestring: Name of the suite.
+        """
+        raise NotImplementedError()
+
+    def benchmarks(self):
+        """Returns the list of the benchmarks provided by this suite.
+
+        Returns:
+            list of strings: List of benchmarks.
+        """
+        raise NotImplementedError()
+
+    def extraDimensions(self, benchmark, vmargs, runargs):
+        """Returns dictionary of extra dimensions for the given benchmark and arguments.
+
+        Arguments:
+            benchmark (string): Name of the benchmark.
+            vmargs (list of strings): List of VM arguments.
+            runargs (list of strings): List of benchmark arguments.
+
+        Returns:
+            dict[basestring, T]: Dictionary of field-name/value pairs.
+        """
+        raise NotImplementedError()
+
+    def run(self, benchmarks, vmargs, runargs):
+        """Runs the specified benchmarks with the given arguments.
+
+        Arguments:
+            benchmarks (list of strings): List of benchmark names.
+            vmargs (list of strings): List of VM arguments.
+            runargs (list of strings): List of benchmark arguments.
+
+        Returns:
+            object: A list of measurement results.
+
+            A measurement result is an object that can be converted into JSON and is
+            merged with root.
```

Comment:
Meant as the root data point object. I'll rewrite it to just `merged with the data point object`.